llvm/include/llvm/Transforms/Utils/LoopUtils.h

```diff
@@ -283,6 +284,22 @@ MDNode *updateIterCounts(LLVMContext &Context, MDNode *LoopID,
                          std::function<int64_t(int64_t)> FixMin,
                          std::function<int64_t(int64_t)> FixMax);
 
+/// Get Minimum Trip Count from the LoopID
+std::optional<int64_t> getMinTripCount(const MDNode *LoopID);
+
+/// Get the LoopID from a single block loop or nullptr
+const MDNode *getLoopID(const MachineBasicBlock &LoopBlock);
+
+/// Get LoopID for Latch
+const MDNode *getLoopID(const BasicBlock *LoopBB);
+
+/// LoopBlock should hold the backedge. Since we generally call it for
+/// single block loops, that's automatically true.
+std::optional<int64_t> getMinTripCount(const MachineBasicBlock &LoopBlock);
```

Comment:
Could you leave helper functions that deal with `MachineBasicBlock` out of this header? I believe it's better to keep LoopInfo about LLVM IR, not MIR.

Reply:
removed the `MachineBasicBlock `from `LoopUtils`